gui/src/ts/components/data/ForceDirectedGraph.tsx

```diff
@@ -10,20 +40,27 @@ export const ForceDirectedGraph = <
   links,
   height,
   width,
-  nodeIdFunction,
-  nodeGroupFunction,
-}: {
+  nodeOptions,
+  linkOptions,
+}: // nodeOptions: userDefinedNodeOptions,
+{
   nodes: NodeDatum[];
   links: LinkDatum[];
   height: number;
   width: number;
-  nodeIdFunction: (node: NodeDatum) => string;
-  nodeGroupFunction: (node: NodeDatum) => string;
+  nodeOptions: NodeOptions<NodeDatum>;
+  linkOptions: LinkOptions<LinkDatum>;
+  // nodeOptions?: Partial<ForceDirectedGraphNodeOptions<NodeDatum>>;
 }) => {
   const svgRef = React.useRef<SVGSVGElement>(null);
```

Comment:
You may want look over this from Uber:
https://github.com/uber/react-vis-force
I used (and modified) it for a project a couple of years ago. It uses React idioms and structure pretty well.

Reply:
Oooh might just switch to that instead, thanks

Reply:
It's dead code that predates hooks, so I'd advise against trying to use it out of the box, but you could adapt it.

Reply:
Okay I'll take a look